test/storybook-playwright/utils.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import type { Page } from '@playwright/test';
+
+const STORYBOOK_PORT = '50241';
+
+export const gotoStoryId = ( page: Page, storyId: string ) =>
+	page.goto(
+		`http://localhost:${ STORYBOOK_PORT }/iframe.html?id=${ storyId }`,
+		{ waitUntil: 'load' }
+	);
+
+export const waitForFMAnimation = async ( { page } ) =>
+	await page.evaluate(
+		() =>
+			new Promise( ( resolve, reject ) => {
+				const rootEl = document.getElementById( 'root' ) as HTMLElement;
+				if ( ! rootEl ) {
+					reject( 'Could not find #root element' );
+				}
+
+				let mutationTimer = setTimeout( resolve, 200 );
+				const observer = new MutationObserver( () => {
+					clearTimeout( mutationTimer );
+					mutationTimer = setTimeout( resolve, 200 );
```

Comment:
Is 200ms the delay during which the animation should happen? What if the animation takes longer?

Reply:
This is the delay between DOM mutations, so I'd say that depends on the animation. If it's an animation that initiates by setting a classname and lasts longer than 200ms then we might have a falstart as there might not be any other subsequent DOM/attribute mutations to debounce of. But from what I understand FM is doing, those attribute mutations should happen much more frequently:

> this is because framer-motion "manually" sets the animated values as inline styles for each frame (instead of relying on native CSS animations and transforms).

If the above is correct, I think that (for FM animations) the attribute mutation observer could be a reliable solution. Does that make sense?

Reply:
I've [updated the approach](https://github.com/WordPress/gutenberg/pull/43393/files/b12a39611068f1e77899df3c5468d07a4dd5fe25..e67b95340e8b54cb823f90741341d3ef871eb889) on this one as it looks like we can use explicit waits for the animated elements. Please see [this comment](https://github.com/WordPress/gutenberg/issues/43282#issuecomment-1225984294) for more context :)